src/Data/Array.purs

```diff
@@ -551,6 +553,32 @@ alterAt i f xs = maybe Nothing go (xs !! i)
 -- Transformations -------------------------------------------------------------
 --------------------------------------------------------------------------------
 
+-- | Inserts the given element in between each element in the array. The array
+-- | must have two or more elements for this operation to take effect.
+-- |
+-- | ```
+-- | intersperse " " [ "a", "b" ] == [ "a", " ", "b" ]
+-- | intersperse 0 [ 1, 2, 3, 4, 5 ] == [ 1, 0, 2, 0, 3, 0, 4, 0, 5 ]
+-- | ```
+-- |
+-- | If the array has one or zero elements, the resulting array is a copy
+-- | of the input array.
```

Comment:
I decided to use copy since it's not technically the "same" array.

Reply:
It actually will be the same array with this implementation. If you mutate one, you’ll see that change reflected in the other.

Reply:
Ah... Shoot, that's right. Let me fix that.

Reply:
Fixed in latest commits

Reply:
I think the previous implementation was preferable - there's no reason to make a copy here, since `Array` is already immutable. You'd have to use an escape hatch like `unsafeThaw` or the FFI to mutate an `Array`, but that's not something we need to defend against, because it can only happen through misuse of said escape hatches.

Reply:
True.
Latest commit reverts that change